### ISSUE TITLE
modDirVersion 5.4.89-hardened specified in the Nix expression is wrong, it should be: 5.4.89.a-hardened

### ISSUE THREAD

**Issue by rokk4** [NixOS/nixpkgs#110302]
**Describe the bug**'
The Kernel update fails with this message.

`...
LD       /build/linux-5.4.89/build/tools/objtool/objtool-in.o
LINK     /build/linux-5.4.89/build/tools/objtool/objtool
make[1]: Leaving directory '/build/linux-5.4.89/build'
Error: modDirVersion 5.4.89-hardened specified in the Nix expression is wrong, it should be: 5.4.89.a-hardened
builder for '/nix/store/m9b2q8p190hg5i1k6sgmv534nrs5qlqf-linux-5.4.89.drv' failed with exit code 1
`

**To Reproduce**
1. Be on Linux 5.4.88-hardened, NixOS, 20.09.2578.d7d6f82e869 
2. nixos-rebuild switch

**Expected behavior**
No errors during update.

**Notify maintainers**
@thoughtpolice

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.88-hardened, NixOS, 20.09.2578.d7d6f82e869 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-20.09.2610.3d8fa5f7215"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
boot.kernelPackages = pkgs.linuxPackages_hardened;
```

**Comment by cole-h** [comment]
cc @NeQuissimus

**Comment by NeQuissimus** [comment]
This is essentially #108707 and should be fixed by 929f40967093bdcdd7ada92c1025879df8e1ea61 for 20.09.
Hardened kernel 5.4.91.a should be coming through to the channel, `nixos-20.09.2660.da3378c4aaf` or later is available on the `nixos-20.09` channel.

Can you try `nixos-rebuild switch --upgrade` and let us know if that works for you?

**Comment by rokk4** [comment]
Autoupgrade run without issues this night, now on 5.4.91.a. , so I guess `nixos-rebuild switch --upgrade` would have worked today as well.
Thanks for your time. <3
